.changeset/fair-jars-behave.md

```diff
@@ -0,0 +1,26 @@
+---
+"astro": minor
+---
+
+Adds a new security - and experimental - option to prevent CSRF attacks. This feature is available only for on-demand pages:
+
+```js
+import { defineConfig } from "astro/config"
+export default defineConfig({
+  security: {
+    csrfProtection: {
+      origin: true
+    }
+  },
+  experimental: {
+    csrfProtection: true
```

Comment:
This sugar version is not explained in the RFC. I'm wondering, should we even have it? I could see arguments that the cookie version is better and should be the default. It might be good to just let them specify which version to use manually.

Reply:
Oh sorry, i this is to enable the protection as an experimental feature. Ignore!